Preview Worker 삭제 파이프라인의 pnpm trust policy 에러 해결

## 목적
PR 종료 시 Preview Worker 자동 삭제 워크플로우에서 발생하는 pnpm trust downgrade 에러 해결

**배경:**
- PR closed 시 Preview Worker 자동 삭제 워크플로우 구현 완료
- wrangler 의존성 설치 중 `ERR_PNPM_TRUST_DOWNGRADE` 에러 발생
- pnpm 10.22.0부터 도입된 trust policy 기능으로 인한 문제
- chokidar@4.0.3 패키지가 provenance(출처 증명) 없이 배포되어 신뢰 수준 하락
- 이전 버전은 provenance attestation이 있었으나 현재 버전은 trust evidence 없음

**에러 상황:**

<img width="1081" height="110" alt="Image" src="https://github.com/user-attachments/assets/94186b6c-5e03-43da-be16-b9956cdb04fd" />

## 작업 내용
- [ ] pnpm trust policy 우회 방법 적용
  - [ ] `--no-strict-peer-dependencies` 플래그 추가 검토
  - [ ] `.npmrc`에 trust 관련 설정 추가
  - [ ] `strict-ssl=false` 또는 trust policy 비활성화 검토
  - [ ] pnpm 10.22.0 이전 버전으로 변경 (trust policy 미적용)
- [ ] 워크플로우 수정 및 테스트

## 예상 효과
- Preview Worker 자동 삭제 워크플로우 정상 작동
- Trust 관련 보안 경고 해결

--- a/.github/workflows/cleanup-preview.yml
+++ b/.github/workflows/cleanup-preview.yml
@@ -3,6 +3,12 @@ name: Cleanup Preview Worker
 on:
   pull_request:
     types: [closed]
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'PR number to clean up'
+        required: true
+        type: number
 
 permissions:
   contents: read
@@ -13,30 +19,29 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Set PR Number
+        id: pr
+        run: |
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            echo "number=${{ github.event.inputs.pr_number }}" >> $GITHUB_OUTPUT
+          else
+            echo "number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+          fi
       
-      - uses: pnpm/action-setup@v3
-        with:
-          version: 10.26.2
-
-      - uses: actions/setup-node@v4
-        with:
-          node-version: '24.12.0'
-          cache: 'pnpm'
-
       - name: Delete Preview Worker
         id: delete
-        uses: cloudflare/wrangler-action@v3
         continue-on-error: true
-        with:
-          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
-          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
-          command: delete --name unibusk-preview-${{ github.event.pull_request.number }}
+        run: npx wrangler@latest delete --name unibusk-preview-${{ steps.pr.outputs.number }}
+        env:
+          CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+          CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
       
       - name: Comment on PR
+        if: github.event_name == 'pull_request'
         uses: actions/github-script@v7
         with:
           script: |
-            const prNumber = context.payload.pull_request.number;
+            const prNumber = ${{ steps.pr.outputs.number }};
             const success = '${{ steps.delete.outcome }}' === 'success';
             const emoji = success ? '✅' : '⚠️';
             const message = success 

